File: src/main/java/core/listeners/GameListener.java
package core.listeners;

import manager.StreamStatusManager;
import net.dv8tion.jda.api.events.user.UserActivityEndEvent;
import net.dv8tion.jda.api.events.user.UserActivityStartEvent;
import net.dv8tion.jda.api.hooks.ListenerAdapter;

public class GameListener extends ListenerAdapter {

    @Override
    public void onUserActivityStart(UserActivityStartEvent event) {
        // Streaming
        StreamStatusManager.onStream(event.getGuild(), event.getMember(), event);
    }

    @Override
    public void onUserActivityEnd(UserActivityEndEvent event) {
        // Streaming
        StreamStatusManager.leaveStream(event.getGuild(), event.getMember(), event);
    }
}

File: src/main/kotlin/core/listeners/GameListener.kt
package core.listeners

import manager.StreamStatusManager.leaveStream
import manager.StreamStatusManager.onStream
import net.dv8tion.jda.api.events.user.UserActivityEndEvent
import net.dv8tion.jda.api.events.user.UserActivityStartEvent
import net.dv8tion.jda.api.hooks.ListenerAdapter


class GameListener : ListenerAdapter() {

    override fun onUserActivityStart(event: UserActivityStartEvent) {
        // Streaming
        onStream(event.guild, event.member, event)
    }

    override fun onUserActivityEnd(event: UserActivityEndEvent) {
        // Streaming
        leaveStream(event.guild, event.member)
    }
}
